JOSS paper: draft (#2939)
openturns/openturns: paper/paper.md

interface, examples, and the underlying probabilistic theory. 
In the Python environment, plotting features are available through the 
Matplotlib module.

## Features
The OpenTURNS library provides functions for defining a probabilistic model 
for a random vector or stochastic process. 
A multivariate distribution can be defined using existing multivariate
distributions, through decomposition into marginals and copulas, or by 
conditioning. 
In the context of a one-dimensional distribution and a given function, it 
is possible to determine the push-forward distribution. 
Furthermore, the library provides support for the definition and composition 
of functions. 
It should be noted that the calculation of the gradient or Hessian of a function 
is taken into account. 
Within the UQ framework, the library plays a pivotal role in facilitating 
the sampling of the input random vector from the probabilistic model. 
This is followed by the propagation of uncertainties through the physical model 
to the output. 
There are a number of sampling methods that can be employed, including Monte 
Carlo, quasi Monte Carlo and variance reduction methods (such as 

> importance sampling and line sampling). 

Reviewer: move line sampling in rare events reliability algos ?

QQ-plot, Kendall plot). 
Nonparametric models can be used based on a sample including kernel 
density estimation or Bernstein copula. 
For a given model, input variables can be ranked using sensitivity analysis 
methods e.g., Standardized Regression Coefficients (SRC), Sobol' and 
Hilbert-Schmidt Independence Criterion (HSIC) indices. 
A function can also be approximated by a meta-model, including linear 
model, polynomial chaos expansion, Gaussian process regression. 
A unified interface to optimization algorithms (NLOpt, Ipopt, PAGMO, Bonmin, 
Dlib, Ceres, Cminpack, Cobyla, etc.) is available and used in particular in 
the context of parametric model calibration with Bayesian or non-Bayesian 
methods. 
Since many probabilistic methods require the evaluation of an integral, 
numerical integration methods are available by sampling or quadrature 
(if necessary, adaptive). 
The provided quadrature methods include multivariate Gaussian quadrature 
(e.g., Gauss–Legendre quadrature), Gauss–Kronrod, Smolyak quadrature, and 
other algorithms including those from the Cuba library. 
The OpenTURNS library provides full or compressed linear algebra capabilities 

> (in particular for covariance matrices). 

Author: This was removed, as you suggested.

propagation, sensitivity analysis, reliability, optimization, inverse
problems, and surrogate modeling.
Validated, well-tested and well-documented methods are essential for teaching, academic research, and industrial applications.
However, many probabilistic modeling libraries are domain-specific, 
focusing on fields such as mechanical engineering, computational fluid
dynamics, or hydrology, which limits the sharing of best practices.
This underscores the need for a general-purpose, cross-disciplinary
framework for probabilistic modeling.

Of course, other Python packages exist for Monte-Carlo
sampling and other numerical methods [@virtanen2020scipy],
statistical modeling [@seabold2010statsmodels], 
machine learning [@pedregosa2011scikit], for providing UQ test 
functions [@Wicaksono2023], and for quasi-Monte Carlo methods [@roy2023quasi]
(among others).
One of the goals of OpenTURNS is to provide a consistent interface to 
these tasks in the UQ context.

# Features

OpenTURNS provides a wide range of multivariate probabilistic modeling
features: more than 60 continuous and discrete distributions, 
including extreme value distributions, copula-based models (more than 

> 15 copulas), and distribution transformations: truncation, conditioning, mixture, push-forward and distribution algebra etc.

Reviewer 2: ```suggestion
15 copulas), and distribution transformations: truncation, conditioning, mixture, push-forward and distribution algebra, etc.
```

for supervised learning or calibration).
OpenTURNS can be used to define a wide variety of optimization problems
in a unified manner and provides users with solution algorithms from
numerous existing libraries: NLOpt, Ipopt, PAGMO, Bonmin, Dlib,
Ceres, Cminpack, Cobyla, and more.
It can solve optimization problems with single
or multiple objectives, with equality or inequality constraints, and
with continuous or discrete variables.

Finally, since many analyses require the computation of integrals,
OpenTURNS provides powerful integration methods based on sampling
or quadrature methods: multivariate Gaussian quadrature (e.g.,
Gauss–Legendre quadrature), Gauss–Kronrod, Smolyak, simplicial cubature,
and other algorithms including those from the Cuba library.

# Usages

OpenTURNS is widely used in research, industrial engineering, and
academic education.
It has been integrated into university curricula (e.g., DTU Denmark) and
engineering schools (e.g., ENSTA, ENPC, Centrale Nantes, and others)
for courses on reliability analysis, risk assessment, and UQ.
Two stable releases are published annually, and the library is

> distributed through Conda, pip, and Debian repositories.

Reviewer 2: ```suggestion
distributed through Conda, pip, and Debian repositories and available in linux, windows and macOs.
```

machine learning [@pedregosa2011scikit], for providing UQ test 
functions [@Wicaksono2023], and for quasi-Monte Carlo methods [@roy2023quasi]
(among others).
One of the goals of OpenTURNS is to provide a consistent interface to 
these tasks in the UQ context.

# Features

OpenTURNS provides a wide range of multivariate probabilistic modeling
features: more than 60 continuous and discrete distributions, 
including extreme value distributions, copula-based models (more than 
15 copulas), and distribution transformations: truncation, conditioning, mixture, push-forward and distribution algebra, etc.
User-defined distributions can be easily implemented and integrated
natively.

Multivariate distributions can be sampled using general sampling algorithms
(Monte-Carlo, Latin Hypercube Sampling - LHS, and optimized LHS) and quasi–Monte Carlo
low-discrepancy sequences (such as Sobol’ or Halton) for integration
and model analysis.

OpenTURNS offers modeling features for stationary 
and non-stationary stochastic processes (ARMA processes, Gaussian 
processes in arbitrary dimensions) associated with several covariance 
kernels (e.g., exponential, Matérn, and squared exponential).

> The library provides extensive data analysis capabilities

Reviewer 3: Issue with the sentence: 
The library provides extensive data analysis capabilities, including the ability to compute dependency indicators

Author: Oups. Ok, I will do this.

Author: You were right: the paragraph was not correctly written. This is fixed.

OpenTURNS can be used to define a wide variety of optimization problems
in a unified manner and provides users with solution algorithms from
numerous existing libraries: NLOpt, Ipopt, PAGMO, Bonmin, Dlib,
Ceres, Cminpack, Cobyla, and more.
It can solve optimization problems with single
or multiple objectives, with equality or inequality constraints, and
with continuous or discrete variables.

Finally, since many analyses require the computation of integrals,
OpenTURNS provides powerful integration methods based on sampling
or quadrature methods: multivariate Gaussian quadrature (e.g.,
Gauss–Legendre quadrature), Gauss–Kronrod, Smolyak, simplicial cubature,
and other algorithms including those from the Cuba library.

# Usages

OpenTURNS is widely used in research, industrial engineering, and
academic education.
It has been integrated into university curricula (e.g., DTU Denmark) and
engineering schools (e.g., ENSTA, ENPC, Centrale Nantes, and others)
for courses on reliability analysis, risk assessment, and UQ.
Stable releases are published on a bi-annual basis, and the library is
distributed through Conda, pip, and Debian repositories and available in 
Linux, Windows and MacOs.

> 

Reviewer 4: @mbaudin47, @m-balesdent , @jschueller , @l-brevault , @sofianehaddad , @regislebrun : could you please validate the references?
I think we can then submit this text.

> 

Author: This suggestion cannot be included because it will make the review process to fail, because it will not match the requirements for a JOSS paper. These are explained at https://joss.readthedocs.io/en/latest/paper.html :

"_The paper should be between 250-1000 words. Authors submitting papers significantly longer than 1000 words may be asked to reduce the length of their paper._"

and some lines after this:

"_Given this format, a “full length” paper is not permitted [...]_"

According to my calculations, the current paper is 946 words (without title, abstract and references), slightly under the threshold. Adding the references you suggest will make it pass the threshold.

> 

Reviewer 3: A compromise to not overshoot the word limitation could be to add just a short sentence such as:

The library has been introduced in the general presentation [@baudin:2016] and used in a broad range of UQ studies across multiple science and engineering domains [quote here two or three references]

The library has been introduced in several publications (e.g.,
[@baudin:2016]).
Its calibration features, for example, are introduced in [@baudin:2021].
The library is also used in several external packages including
the benchmarking module otbenchmark [@fekhari:2021].
Numerous application articles are based on OpenTURNS (e.g., [@Dutfoy:2023]
on seismic risk assessment using the Gutenberg–Richter law).

The Persalys graphical user interface [@persalys], built on top of OpenTURNS,
enables users to perform UQ analyses with no (or minimal) Python
programming.

An annual Users’ Day has been held since 2008, fostering collaboration
and knowledge exchange among researchers, engineers, and educators.

> # References

Reviewer 4: We should add some references coming from Onera, Airbus and Phimeca....
@m-balesdent , @jschueller , @l-brevault , @sofianehaddad , @regislebrun : could you please add some publications and courses using OpenTURNS?
I think we can then submit this text.

However, many probabilistic modeling libraries are domain-specific, 
focusing on fields such as mechanical engineering, computational fluid
dynamics, or hydrology, which limits the sharing of best practices.
This underscores the need for a general-purpose, cross-disciplinary
framework for probabilistic modeling.

Of course, other Python packages exist for Monte-Carlo
sampling and other numerical methods [@virtanen2020scipy],
statistical modeling [@seabold2010statsmodels], 
machine learning [@pedregosa2011scikit], for providing UQ test 
functions [@Wicaksono2023], and for quasi-Monte Carlo methods [@roy2023quasi]
(among others).
One of the goals of OpenTURNS is to provide a consistent interface to 
these tasks in the UQ context.

# Features

OpenTURNS provides a wide range of multivariate probabilistic modeling
features: more than 60 continuous and discrete distributions, 
including extreme value distributions, copula-based models (more than 
15 copulas), and distributions defined by conditioning.
It is also possible to determine exactly any univariate push-forward
distribution (for example, the exact distribution of the cosine of a

> distribution).

Reviewer: ```suggestion
including extreme value distributions, copula-based models (more than 
15 copulas), and distribution transformations: truncation, conditioning, mixture, push-forward and distribution algebra etc.
```

User-defined distributions can be easily implemented and integrated
natively.

Multivariate distributions can be sampled using general sampling algorithms
(Monte Carlo, LHS, and optimized LHS) and quasi–Monte Carlo
low-discrepancy sequences (such as Sobol’ or Halton) for integration
and model analysis.

OpenTURNS offers modeling features for stationary 
and non-stationary stochastic processes (ARMA processes, Gaussian 
processes in arbitrary dimensions) associated with several covariance 
kernels (e.g., exponential, Matèrn, and squared exponential).

OpenTURNS provides numerous possibilities for multivariate data analysis.
Dependence indicators can be computed, such as Pearson and
Spearman correlations.
Furthermore, OpenTURNS can fit a multivariate 
distribution to a sample using parametric and non-parametric
methods.
Copula estimation can also be performed using a copula model or the
non-parametric Bernstein copula.
OpenTURNS provides many tests to validate the estimation: quantitative
tests such as the Kolmogorov–Smirnov or Lilliefors tests, and

> graphical tests such as the quantile–quantile plot.

Reviewer: Here is a proposal to rephrase it in a more compact way

Several sensitivity and reliability indices can be used to quantify
how influential a variable is to another variable.
Notable examples include Standardized Regression Coefficients (SRC),
Sobol’ indices, and Hilbert–Schmidt Independence Criterion (HSIC) indices.
For the latter two, OpenTURNS provides the asymptotic distribution of
the estimators, which allows the computation of confidence intervals.

OpenTURNS implements numerous reliability algorithms for rare-event
probabilities: classic algorithms such as the First-Order Reliability
Method, Second-Order Reliability Method, subset sampling, and other
variance reduction methods, as well as more recent ones such as
Nonparametric Adaptive Importance Sampling (NAIS) and Cross-Entropy.
An event can be a single event (for example, threshold exceedance), a
system event, or an event based on a stochastic process.

For supervised learning, OpenTURNS enables the construction
of surrogate models from data, such as linear regression models,
polynomial chaos expansion, and Gaussian process regression.
OpenTURNS provides polynomial and functional orthogonal bases such 

> as Haar wavelets or Fourier series.

Reviewer: I propose to drop this to keep the text short

This surrogate model can also be built for models with functional inputs.
The optimization of the hyperparameters of Gaussian process regression
can be performed on an arbitrary subset of the hyperparameters.

In many cases, probabilistic analyses use optimization algorithms (e.g.,
for supervised learning or calibration).
OpenTURNS can be used to define a wide variety of optimization problems
in a unified manner and provides users with solution algorithms from
numerous existing libraries: NLOpt, Ipopt, PAGMO, Bonmin, Dlib,
Ceres, Cminpack, Cobyla, and more.
It can solve optimization problems with single
or multiple objectives, with equality or inequality constraints, and
with continuous or discrete variables.

Finally, since many analyses require the computation of integrals,
OpenTURNS provides powerful integration methods based on sampling
or quadrature methods: multivariate Gaussian quadrature (e.g.,
Gauss–Legendre quadrature), Gauss–Kronrod, Smolyak, simplicial cubature,

> and other algorithms including those from the Cuba library.

Author: Removing this sentence means that integration methods do not appear in the paper at all. It seems to me that this produces a less interesting paper for readers interested by this topic, which is strongly related to many probabilistic analyses. Finally, I think that the paper length is now within the required limits, so that there is no particular reason to make it shorter.

---
title: 'OpenTURNS: A Python package for uncertainty quantification'
tags:
  - Python
  - uncertainty quantification
  - probabilistic modeling
  - statistics
  - meta-modeling
  - calibration
  - stochastic processes
  - Bayesian modeling
  - machine learning
  - supervised machine learning
  - unsupervised machine learning
  - surrogate
authors:
  - name: Michaël Baudin
    orcid: 0000-0002-0450-7947
    affiliation: 1
  - name: Anne Dutfoy
    affiliation: 1
  - name: Joseph Muré
    affiliation: 1
  - name: Julien Pelamatti
    affiliation: 1
  - name: Régis Lebrun
    affiliation: 2
  - name: Sofiane Haddad
    affiliation: 2
  - name: Julien Schueller
    affiliation: 3
  - name: Loïc Brevault
    affiliation: 4
  - name: Mathieu Balesdent
    affiliation: 4
affiliations:
 - name: EDF R&D, France
   index: 1
 - name: Airbus, France
   index: 2
 - name: Phimeca, France
   index: 3
 - name: ONERA, France
   index: 4
date: 30 April 2025
bibliography: paper.bib
---

# Summary

``OpenTURNS`` is a C++ library for uncertainty propagation by probabilistic methods.
OpenTURNS is also available as a Python module. 

## Introduction

> Numerical simulation and the uncertainties associated with these models make decision-making robust in the design and operation of complex systems. Supervisory authorities in safety-critical industrial sectors (e.g. nuclear power or civil aviation) are demanding that players use tried-and-tested methods, backed up by tested, documented and auditable implementations. As a result, several uncertainty processing libraries have emerged over the last few decades. However, many of these approaches are dedicated to a particular discipline, such as numerical fluid mechanics or hydrology. This approach limits the sharing of best practices. This is why a generic approach to uncertainty management is so important.

Reviewer 3: Some minor suggestions : 
-  "Numerical simulation and the uncertainties associated with these models make decision-making robust in the design and operation of complex systems." --> Numerical simulation and the uncertainties inherent in these models play a crucial role in supporting robust decision-making for the design and operation of complex systems.
- "are demanding that players use tried-and-tested methods" --> require the use of proven methods supported by well-tested, documented, and auditable implementations. 
- "This is why a generic approach to uncertainty management is so important." --> This highlights the need for a generic, cross-disciplinary approach to uncertainty management.

## Overview
OpenTURNS is an open-source software under the LGPL license based on a C++ library and provides a Python package. OpenTURNS is a generic library for probabilistic modeling and uncertainty quantification. The library provides full documentation, in particular of the programming interface, examples and probabilistic theory. In Python, plotting features are provided using Matplotlib. 

> ## Features

Reviewer 3: Should we add somewhere in the Features that it is possible to "wrap/interface with" any "simulation model/codes" that can be evaluated in Python ? 
It can be important to the interested user to understand that it is possible to carry out UQ with OpenTURNS on any code/model if a standard "wrapping" is carried out (through PythonFunction or otwrapy) ?

Author: Very true. I added a paragraph on this crucial topic.

> ## Usages

Reviewer 3: Maybe in Usages (or somewhere else in the paper) it could be valuable to state that the release cycle is twice per year ?

> The OpenTURNS library is used for research and engineering studies. OpenTURNS is used for higher education at universities (DTU Denmark) and engineering schools (ENSTA, ENPC, Centrale Nantes, Sigma Clermont, Sup Galilée).

Author: Thank you for this comment. This is fixed.

The library is presented in several publications, e.g. `@Baudin:2016`. For example, calibration features of the library are introduced in `@Baudin:2021`. External modules such as the benchmarking module `otbenchmark` is introduced in `@Fekhari:2021`. Numerous application articles are based on OpenTURNS (e.g. `@Dutfoy:2023` on seismic risk assessment using the Gutenberg-Richter law).

The Persalys graphical user interface provides uncertainty quantification features using OpenTURNS (see www.persalys.fr) . Other Persalys-derived graphical tools are also based on OpenTURNS (see `@Bryla:2022`). 

> A User's day meeting has been organized each year since 2008. The goal of this meeting is to present the use of OpenTURNS in engineering or research uncertainty quantification studies.

Reviewer 3: it could be interesting to say that we also present to the community the novelty in terms of developments since the previous User's day edition.
for instance :
"of this meeting is" to introduce to the community the new developments from the previous releases and to present

Author: I added a new sentence: "It's also an opportunity to present the new features of each version of the library, as well as the roadmap.". So this is fixed.

---
title: 'OpenTURNS: A Python package for uncertainty quantification'
tags:
  - Python
  - uncertainty quantification
  - probabilistic modeling
  - statistics
  - meta-modeling
  - calibration
  - stochastic processes
  - Bayesian modeling
  - machine learning
  - supervised machine learning
  - unsupervised machine learning
  - surrogate
authors:
  - name: Michaël Baudin
    orcid: 0000-0002-0450-7947
    affiliation: 1
  - name: Anne Dutfoy
    affiliation: 1
  - name: Joseph Muré
    affiliation: 1
  - name: Julien Pelamatti
    affiliation: 1
  - name: Régis Lebrun
    affiliation: 2
  - name: Sofiane Haddad
    affiliation: 2
  - name: Julien Schueller
    affiliation: 3
  - name: Loïc Brevault
    affiliation: 4
  - name: Mathieu Balesdent
    affiliation: 4
affiliations:
 - name: EDF R&D, France
   index: 1
 - name: Airbus, France
   index: 2
 - name: Phimeca, France
   index: 3
 - name: ONERA, France
   index: 4
date: 30 April 2025
bibliography: paper.bib
---

# Summary

``OpenTURNS`` is a C++ library for uncertainty propagation by probabilistic methods.
OpenTURNS is also available as a Python module. 

## Introduction

> Uncertainties in numerical simulations and play a crucial role in enabling robust decision-making for the design and operation of complex systems. Supervisory authorities in safety-critical industrial sectors (e.g., nuclear power or civil aviation) require the use of validated methods, supported by well-tested, documented, and auditable implementations. As a result, several uncertainty quantification (UQ) libraries have emerged over the past few decades. However, many of these tools are tailored to specific disciplines, such as mechanical engineering, computational fluid dynamics or hydrology, which limits the sharing of best practices. This underscores the need for a generic, cross-disciplinary approach to UQ.

Reviewer 5: Fixed

## Overview
OpenTURNS is an open-source software released under the LGPL license. It is built on a C++ library and provides a Python package. OpenTURNS is a generic library for probabilistic modeling and UQ. The library offers comprehensive documentation, including the programming interface, examples, and the underlying probabilistic theory. In Python, plotting features are available through Matplotlib.

## Features

> The OpenTURNS library provides functions for defining a probabilistic model for a random vector or stochastic process. A multivariate distribution can be defined in  several ways: directly, by decomposition into marginals and copulas, or by conditioning. For a one-dimensional distribution and a given function, we can determine the push-forward distribution. In addition, the library supports the definition and composition of functions. The calculation of the gradient or hessian of a function is taken into account. In the UQ context, the library provides services for sampling the distribution of the input random vector from the probabilistic model and propagating uncertainties through the physical model  to the output. Sampling can be performed using Monte-Carlo, quasi Monte Carlo and variance reduction methods (importance sampling, line sampling). Different types of experimental designs are implemented including Latin Hypercube Sampling (LHS), optimized LHS and low-discrepency sequences (e.g., Sobol', Faure, Halton).

Author: I do not agree with the suggestion (Github does not help to see the differences, but https://www.diffchecker.com/fr/text-compare/ does). 
- A multivariate distribution can be defined using 3 different methods : "directly" means for a distribution which is already multidimensional. This option seems to be removed in your suggestion. Is that on purpose?
- The sentence : "Please be advised that the calculation of the gradient or Hessian of a function is taken into account." seems strange to me. We generally do not directly talk to the read this way in a formal paper.

> The OpenTURNS library provides functions for defining a probabilistic model for a random vector or stochastic process. A multivariate distribution can be defined in  several ways: directly, by decomposition into marginals and copulas, or by conditioning. For a one-dimensional distribution and a given function, we can determine the push-forward distribution. In addition, the library supports the definition and composition of functions. The calculation of the gradient or hessian of a function is taken into account. In the UQ context, the library provides services for sampling the distribution of the input random vector from the probabilistic model and propagating uncertainties through the physical model  to the output. Sampling can be performed using Monte-Carlo, quasi Monte Carlo and variance reduction methods (importance sampling, line sampling). Different types of experimental designs are implemented including Latin Hypercube Sampling (LHS), optimized LHS and low-discrepency sequences (e.g., Sobol', Faure, Halton).

Author: Everything seems right to me, except the second sentence which is long and awkward in my opinion. I would suggest this:
```suggestion
The OpenTURNS library provides functions for defining a probabilistic model for a random vector or stochastic process. A multivariate distribution can be defined using existing multivariate distributions, through decomposition into marginals and copulas, or by conditioning. In the context of a one-dimensional distribution and a given function, it is possible to determine the push-forward distribution. Furthermore, the library provides support for the definition and composition of functions. It should be noted that the calculation of the gradient or Hessian of a function is taken into account. Within the UQ framework, the library plays a pivotal role in facilitating the sampling of the input random vector from the probabilistic model. This is followed by the propagation of uncertainties through the physical model to the output. There are a number of sampling methods that can be employed, including Monte Carlo, quasi Monte Carlo and variance reduction methods (such as importance sampling and line sampling). A variety of experimental designs are employed, encompassing Latin Hypercube Sampling (LHS), Optimized LHS and low-discrepancy sequences (e.g., Sobol', Faure, Halton).
```
Is that OK for you?

> The OpenTURNS library provides functions for defining a probabilistic model for a random vector or stochastic process. A multivariate distribution can be defined in  several ways: directly, by decomposition into marginals and copulas, or by conditioning. For a one-dimensional distribution and a given function, we can determine the push-forward distribution. In addition, the library supports the definition and composition of functions. The calculation of the gradient or hessian of a function is taken into account. In the UQ context, the library provides services for sampling the distribution of the input random vector from the probabilistic model and propagating uncertainties through the physical model  to the output. Sampling can be performed using Monte-Carlo, quasi Monte Carlo and variance reduction methods (importance sampling, line sampling). Different types of experimental designs are implemented including Latin Hypercube Sampling (LHS), optimized LHS and low-discrepency sequences (e.g., Sobol', Faure, Halton).

Author: @sofianehaddad : Can you please review this suggestion?

## Introduction
Uncertainties in numerical simulations play a crucial role in enabling robust decision-making for the design and operation of complex systems. Supervisory authorities in safety-critical industrial sectors (e.g. nuclear power or civil aviation) require the use of validated methods, supported by well-tested, documented, and auditable implementations. Consequently, there has been significant growth in the development of uncertainty quantification (UQ) libraries in recent decades. However, many of these tools are tailored to specific disciplines, such as mechanical engineering, computational fluid dynamics or hydrology, which limits the sharing of best practices. This emphasises the need for a generic, cross-disciplinary approach to UQ.

## Overview
OpenTURNS is a generic library for probabilistic modelling and UQ, released under the LGPL license. The software is built on a C++ library and provides a Python package. The library provides comprehensive documentation, including the programming interface, examples, and the underlying probabilistic theory. In the Python environment, plotting features are available through the Matplotlib module.

## Features

> The OpenTURNS library provides functions for defining a probabilistic model for a random vector or stochastic process. A multivariate distribution can be defined using existing multivariate distributions, through decomposition into marginals and copulas, or by conditioning. In the context of a one-dimensional distribution and a given function, it is possible to determine the push-forward distribution. Furthermore, the library provides support for the definition and composition of functions. It should be noted that the calculation of the gradient or Hessian of a function is taken into account. Within the UQ framework, the library plays a pivotal role in facilitating the sampling of the input random vector from the probabilistic model. This is followed by the propagation of uncertainties through the physical model to the output. There are a number of sampling methods that can be employed, including Monte Carlo, quasi Monte Carlo and variance reduction methods (such as importance sampling and line sampling). A variety of experimental designs are employed, encompassing Latin Hypercube Sampling (LHS), Optimized LHS and low-discrepancy sequences (e.g., Sobol', Faure, Halton).

Reviewer: J'aurais mentionné qu'il y a des distrbutions continues et discrètes.
À la place de la push-forward seulement j'aurais énuméré rapidement les transformations de distribution courantes: jointe, truncation, mixture, random-mixture, push-forward, marginale, conditionnelles... et éventuellement mentionné qu'il y a aussi des transformations pour les copules.

> When the physical model is available as a Python function, using the library is straightforward. However, evaluating the model directly in Python is not always easy. For physical models that can be executed from the operating system's command line, the library provides coupling tools that wrap the executable as a Python function, using text files for input and output.

Reviewer: For physical models->codes
